Algorithm.Framework/Alphas/RsiAlphaModel.cs

```diff
@@ -0,0 +1,275 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Indicators;
+using QuantConnect.Util;
+
+namespace QuantConnect.Algorithm.Framework.Alphas
+{
+    /// <summary>
+    /// Uses Wilder's RSI to create insights. Using default settings, a cross over below 30 or above 70 will
+    /// trigger a new insight.
+    /// </summary>
+    public class RsiAlphaModel : IAlphaModel
+    {
+        private readonly Parameters _parameters;
+        private readonly Dictionary<Symbol, SymbolData> _symbolDataBySymbol = new Dictionary<Symbol, SymbolData>();
+
+        /// <summary>
+        /// Initializes a new default instance of the <see cref="RsiAlphaModel"/> class.
+        /// This uses the traditional 30/70 bounds coupled with 5% bounce protection.
+        /// The traditional period of 14 days is used and the prediction interval is set to 14 days as well.
+        /// </summary>
+        public RsiAlphaModel()
+            : this(new Parameters())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RsiAlphaModel"/> class
+        /// </summary>
+        /// <param name="parameters">Model parameters</param>
+        public RsiAlphaModel(Parameters parameters)
+        {
+            _parameters = parameters;
+        }
+
+        /// <summary>
+        /// Updates this alpha model with the latest data from the algorithm.
+        /// This is called each time the algorithm receives data for subscribed securities
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="data">The new data available</param>
+        /// <returns>The new insights generated</returns>
+        public IEnumerable<Insight> Update(QCAlgorithmFramework algorithm, Slice data)
```

Comment:
Just for sake of ease of understanding even though it supports IEnumerable<> I wonder if better to keep the demos to a List<> or Array

Reply:
Updated implementation to use `List<Insight>` instead of `yield return` syntax